src/catalog/src/local/manager.rs

```diff
@@ -427,19 +427,23 @@ impl CatalogManager for LocalCatalogManager {
             })?;
 
         let engine = old_table.table_info().meta.engine.to_string();
-        // rename table in system catalog
-        self.system
-            .register_table(
-                catalog_name.clone(),
-                schema_name.clone(),
-                request.new_table_name.clone(),
-                request.table_id,
-                engine,
-            )
-            .await?;
-        Ok(schema
-            .rename_table(&request.table_name, request.new_table_name)
-            .is_ok())
+
+        let renamed = schema
+            .rename_table(&request.table_name, request.new_table_name.clone())
+            .is_ok();
+        if renamed {
+            // rename table in system catalog
+            self.system
```

Comment:
correctness: register_table may fail, then the state in memory is inconsistent with the system table, so I think the correct steps are:
- acquire the register_lock to avoid other threads registering a table with the same new name
- check whether the table with the new name exists
- register the table to the system catalog table (overwrite the table entry)
- rename the table in the schema (should always success)